experimental/sql/src/projection_optimizations.py

```diff
@@ -0,0 +1,191 @@
+    elif isinstance(op, RelAlg.Aggregate):
+      rewriter.replace_matched_op(
+          RelAlg.Aggregate.get(Region.from_operation_list([new_proj]),
+                               [s.data for s in op.col_names.data],
+                               [s.data for s in op.functions.data],
+                               [s.data for s in op.res_names.data]))
+
+
+@dataclass
+class ProjectionFuser(ProjectionOptimizer):
+
+  # This rewriter fuses two subsequent projections. Currently, it only works on
+  # renaming style projections, but it will be extended in the future.
+
+  def rename_expr(self, op: RelAlg.Expression, map: dict[str,
+                                                         str]) -> Operation:
+    if isinstance(op, RelAlg.Column):
+      return RelAlg.Column.get(map[op.col_name.data])
+    if isinstance(op, RelAlg.Multiply):
+      return RelAlg.Multiply.get(
+          Region.from_operation_list([self.rename_expr(op.lhs, map)]),
+          Region.from_operation_list([self.rename_expr(op.rhs, map)]))
+    if isinstance(op, RelAlg.Literal):
+      return RelAlg.Literal.get(op.val, op.type)
+    if isinstance(op, RelAlg.Compare):
+      return RelAlg.Multiply.get(
+          op.comparator.data,
+          Region.from_operation_list([self.rename_expr(op.left, map)]),
+          Region.from_operation_list([self.rename_expr(op.right, map)]))
+
+  @op_type_rewrite_pattern
+  def match_and_rewrite(self, op: RelAlg.Project, rewriter: PatternRewriter):
+    if not isinstance(op.input.op, RelAlg.Project):
+      return
+
+    child_cols = self.flatten(
+        [self.find_cols_in_expr(o) for o in op.input.op.projections.ops])
+    child_res = [s.data for s in op.input.op.names.data]
+    # If the child projection simply projects onto a subset, we can just infer
+    # its column mapping and apply it on the parents projection.
+    if len(child_cols) == len(child_res) and all(
+        [isinstance(o, RelAlg.Column) for o in op.input.op.projections.ops]):
+      child_dict = dict(zip(child_cols, child_res))
+
+      rewriter.replace_matched_op(
+          RelAlg.Project.get(
+              rewriter.move_region_contents_to_new_regions(op.input.op.input),
+              Region.from_operation_list([
+                  self.rename_expr(o, child_dict) for o in op.projections.ops
+              ]), op.names))
+
+
+@dataclass
+class IdentityProjectionRemover(ProjectionOptimizer):
+
+  @op_type_rewrite_pattern
+  def match_and_rewrite(self, op: RelAlg.Project, rewriter: PatternRewriter):
+
+    # If any of the expressions in the projections list is not a Column
+    # expression, this is not an IdentityProjection.
+    # TODO: Maybe optimize simple expressions paterns (like add 0 or mult 1)
+    if not all([isinstance(o, RelAlg.Column) for o in op.projections.ops]):
+      return
+
+    input_schema = self.find_schema(op.input.op)
+    output_schema = self.flatten(
+        [self.find_cols_in_expr(o) for o in op.projections.ops])
```

Comment:
It should be possible to inspect get the column names directly, since we know that all `projections.ops` are `Column` ops, no?